src/main/Sirius.java

```diff
@@ -0,0 +1,291 @@
+        }
+        System.out.println(SEPARATOR);
+    }
+    public static String[] splitCommand(String userInput) throws InvalidTaskContentException, IncompleteCommandException{
+        // It returns a commandPiece, which contents prefix, taskName, by time for deadline, and from/to time for event.
+        String[] slashCommand = userInput.split(SLASH);
+        String commandContent = slashCommand[0]; //commandContent = commandPrefix + taskName
+        String commandPrefix = commandContent.split(SPACE)[0];
+        String taskName = commandContent.replace(commandPrefix, EMPTY).trim();
+
+        String[] commandPieces = {commandPrefix, taskName, EMPTY, EMPTY};
+        try {
+            switch (commandPrefix) {
+                case MARK:
+                case UNMARK:
+                    if (taskName.isEmpty()){
+                        isValidToProcess = false;
+                        throw new IncompleteCommandException("task index");
+                    }
+                    break;
+                case TODO:
+                     if (taskName.isEmpty()) {
+                         isValidToProcess = false;
+                         throw new IncompleteCommandException(commandPrefix);
+                     }
+                     break;
+                case DEADLINE:
+                     int indexOfBy = userInput.indexOf("/by");
+                     if (taskName.isEmpty()) { //list or mark/task empty
+                         isValidToProcess = false;
+                         throw new IncompleteCommandException(commandPrefix);
+                     }
+                     else if (indexOfBy == -1) {
+                         isValidToProcess = false;
+                         throw new InvalidTaskContentException("You should declare '/by' for deadline");
+                     }
+                     else {
+                         commandPieces[2] = slashCommand[1].replace("by", EMPTY).trim();
+                         return commandPieces;
+                     }
+                case EVENT:
+                    int indexOfFrom = userInput.indexOf("/from");
+                    int indexOfTo = userInput.indexOf("/to");
+                    if (taskName.isEmpty()) { //list or mark/task empty
+                        isValidToProcess = false;
+                        throw new IncompleteCommandException(commandPrefix);
+                    }
+                    else if (indexOfFrom == -1 || indexOfTo == -1) {
+                        isValidToProcess = false;
+                        throw new InvalidTaskContentException("You should declare '/from' and '/to' for event");
+                    }
+                    else {
+                        commandPieces[2] = slashCommand[1].replace("from", EMPTY).trim();
+                        commandPieces[3] = slashCommand[2].replace("to", EMPTY).trim();
+                        return commandPieces;
+                    }
+                }
```

Comment:
Consider  breaking it down into methods like `processTodoCommand`, `processDeadlineCommand`, etc.